### ISSUE TITLE
Getting de.timroes.axmlrpc.XMLRPCServerException: parse error. not well formed exception.

### ISSUE THREAD

**Issue by thebhushanp** [gturri/dokuJClient#27]
I getting following exception when I am trying to upload page to DokuWiki.
`at java.lang.Thread.run(Thread.java:745)
Caused by: dw.xmlrpc.exception.DokuUnknownException: de.timroes.axmlrpc.XMLRPCServerException: parse error. not well formed [-32700]
	at dw.xmlrpc.ExceptionConverter.Convert(ExceptionConverter.java:105)
	at dw.xmlrpc.CoreClient.genericQuery(CoreClient.java:49)
	at dw.xmlrpc.DokuJClient.genericQuery(DokuJClient.java:702)
	at dw.xmlrpc.DokuJClient.putPage(DokuJClient.java:500)
	at dw.xmlrpc.DokuJClient.putPage(DokuJClient.java:472)
	at com.paramatrix.wikimigration.apis.doku.DokuService.addPage(DokuService.java:34)
	... 6 more
Caused by: de.timroes.axmlrpc.XMLRPCServerException: parse error. not well formed [-32700]
	at de.timroes.axmlrpc.ResponseParser.parse(ResponseParser.java:78)
	at de.timroes.axmlrpc.XMLRPCClient$Caller.call(XMLRPCClient.java:744)
	at de.timroes.axmlrpc.XMLRPCClient.call(XMLRPCClient.java:461)
	at dw.xmlrpc.CoreClient.genericQuery(CoreClient.java:43)
	... 10 more`

Content of page I uploading is in below mentioned file.
[2.dw.zip](https://github.com/gturri/dokuJClient/files/816647/2.dw.zip)

**Comment by gturri** [comment]
Thank you for this bug report.

I'm going to need some more info. Could you please tell me:

* the version of Dokuwiki you are using
* the version of Dokujclient you are using
* do you have php errors in the logs of your server?
* is your Dokuwiki configured to use English, or another langage? (error identification might not work properly for wiki not in English)
* are you able to use dokujclient for other queries, or does it fail all the time?

**Comment by thebhushanp** [comment]
Hi @gturri 

- I am using '2016-06-26a "Elenor of Tsort" ' version of DokuWiki.
- I am using Dokujclient of version 3.9.0
- Where can I find php errors? 
- Dokuwiki configured to use English.
- Yes, I am able to use dokujclient for other queries.

I tried to upload same content using [Python client](http://python-dokuwiki.readthedocs.io/en/latest/) and it worked. So my guess is that this issue has something to do with Dokujclient not with DokuWiki XML-RCP module.

**Comment by gturri** [comment]
Thanks.

The place where you can find potential php errors depends no your server. For instance, I'm using apache on Ubuntu, and I would find the logs in `/var/log/apache2/error.log`. (And when I was using Easyphp on Windows, I think I needed to right click on the Easyphp icon, and then select `log` > `Apache`).

Anyway,  something else that would bring additional information, would be to build an instance of `Dokujclient` with the `debug` flag on. So that it will print on `System.out` the response received from the server.

To do it, instead of building an instance with `new DokuJClient(url, user, pwd)`, you can do:

    DokuJClientConfig config = new DokuJClientConfig(url);
    config.setUser(user, pwd);
    config.setDebug(true);
    new DokuJClient(config);

It would be useful if you can do it, because I'm not able to reproduce this issue. Actually, I don't get any exception when I run the following code:

	public static void main(String[] args) throws Exception {
		DokuJClientConfig config = new DokuJClientConfig(<url>);
		config.setUser(<user>, <pwd);
		config.setDebug(true);
		DokuJClient dokujclient = new DokuJClient(config);			
		dokujclient.putPage("somePage", getString());
    }
	
	private static String getString(){
		return "" +
				"====== test ======" +
				"" +
				"<code>" +
				"<![CDATA[AI-Za-z0-9-_)(}{!@#$%&<>/]]]" +
				"</code>";
	}

which prints:

    [login eluded for brievety]
    <methodCall>
    <methodName>wiki.putPage</methodName>

    <params>
    <param>
    <value>
    <string>somePage</string>
    </value>
    </param>

    <param>
    <value>
    <string>====== test ======&lt;code>&lt;![CDATA[AI-Za-z0-9-_)(}{!@#$%&amp;&lt;>/]]]&lt;/code></string>
    </value>
    </param>

    <param>
    <value>
    <struct/>
    </value>
    </param>
    </params>
    </methodCall>

    <?xml version="1.0" encoding="UTF-8" standalone="no"?>
    <methodResponse>
      <params>
        <param>
          <value>
            <boolean>1</boolean>
          </value>
        </param>
      </params>
    </methodResponse>

**Comment by thebhushanp** [comment]
Hi @gturri , I  don't see any errors in /var/log/apache2/error.log
Are you able to upload following content?
```
private static String getString() {
        return "\\ Page was created by Bhushan .\\ "
            + "Created on:  Mon Jun 19 15:48:55 IST 2000"
            + "<code> <![CDATA[ ]]> </code>";
    }
```
I am getting following messages on console
```
<?xml version="1.0" encoding="UTF-8"?>

<methodCall>
<methodName>dokuwiki.login</methodName>

<params>
<param>
<value>
<string>myuser</string>
</value>
</param>

<param>
<value>
<string>mypassword</string>
</value>
</param>
</params>
</methodCall>

<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<methodResponse>
  <params>
    <param>
      <value>
        <boolean>1</boolean>
      </value>
    </param>
  </params>
</methodResponse>
<?xml version="1.0" encoding="UTF-8"?>

<methodCall>
<methodName>wiki.putPage</methodName>

<params>
<param>
<value>
<string>test2</string>
</value>
</param>

<param>
<value>
<string>\ Page was created by karthike.\ Created on:  Mon Jun 19 15:48:55 IST 2000&lt;code> &lt;![CDATA[ ]]> &lt;/code></string>
</value>
</param>

<param>
<value>
<struct/>
</value>
</param>
</params>
</methodCall>

<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<methodResponse>
  <fault>
    <value>
      <struct>
        <member>
          <name>faultCode</name>
          <value>
                        <int>-32700</int>
                    </value>
        </member>
        <member>
          <name>faultString</name>
          <value>
                        <string>parse error. not well formed</string>
                    </value>
        </member>
      </struct>
    </value>
  </fault>
</methodResponse>
Mar 06, 2017 9:48:01 AM dw.xmlrpc.CoreClient genericQuery
INFO: Caught exception when executing action wiki.putPage: de.timroes.axmlrpc.XMLRPCServerException: parse error. not well formed [-32700]
```

**Comment by gturri** [comment]
With this string I indeed reproduce the error. I'm going to have a deeper look.

**Comment by thebhushanp** [comment]
Hi @gturri , any luck finding this issue?

**Comment by gturri** [comment]
Yes: it is a bug in the underlying xmlrpc library. I've just fixed this issue: https://github.com/gturri/aXMLRPC/commit/5c8d8dfdfe559cccb3e0a06b707b223c61d22339 ; and the artifact is being uploaded to maven central. Once the upload is done, I'll just to update the `pom.xml` of dokujclient in order to use v1.10.0 of aXMLRPC and it should be good.
Should be ok in the next few days I guess.

**Comment by thebhushanp** [comment]
If I need to make it work now then I guess, I need to build aXMLRPC dependency from source code. Right ?

**Comment by gturri** [comment]
aXMLRPC 1.10.0 is available on maven central (http://search.maven.org/#artifactdetails%7Cfr.turri%7CaXMLRPC%7C1.10.0%7Cjar), so you don't need to build it yourself

**Comment by gturri** [comment]
It is fixed in dokujclient 3.9.1 (which should be available in maven Central in the next few hours).

Thanks for your help in identifying this bug!
